### ISSUE TITLE
Restrict nicknames

### ISSUE THREAD

**Issue by olegmisar** [haxball/haxball-issues#636]
Players' nicknames restriction will make developers' life easier. Sometimes players use only emoji to form a nickname or characters which are hard are hardly recognizable not only by script but even by people. By now, I have to consider different languages, emoji, special characters etc to pick players by nickname. _Picking by id is possible but nobody uses it._ So, I suggest to:

- restrict players' nicknames to only English characters, digits and underline character "_"
- set minimum length of nickname to 2 characters (3 characters will produce more unique nicknames but 2 is fine too)

Also I want to get feedback from other players from community to make this issue better. 

I know it is hard to integrate this restrictions into haxball now because there are a lot of players with different nicknames but we can make a list of allowed characters bigger.

**Comment by olegmisar** [comment]
Yeah, my picker is very good and handle all nicknames except nicknames made of emoji or non-ascii characters but it was very hard to implement it.
Nicknames restriction is a common practise in games. So, why don't we have this in haxball?

**Comment by NiExTy** [comment]
> Nicknames restriction is a common practise in games. So, why don't we have this in haxball?

Likely because we do not have account system, and players want to distinguish themselves from fakes, and similar names. For example 2 people can have one username just styled differently.
